test/lib/service/oauth2_service_test.exs

```diff
@@ -33,5 +36,59 @@ defmodule EaRestaurantDataLoader.Oauth2ServiceTest do
       assert login_response.expires_in == app_client.access_token_expiration_time
       assert login_response.scopes == scopes.scope
     end
+
+    test " refresh token when expired access token" do
```

Comment:
* Add a test when a new access token is not build.
* Add a test where the refresh token is not valid anymore

Reply:
Done.